lib/ansible/executor/playbook_executor.py

```diff
@@ -247,6 +247,9 @@ def _generate_retry_inventory(self, retry_path, replay_hosts):
         re-running on ONLY the failed hosts.  This may duplicate some variable
         information in group_vars/host_vars but that is ok, and expected.
         '''
+        dirname = os.path.dirname(retry_path)
+        if not os.path.exists(dirname):
+            os.makedirs(dirname)
```

Comment:
there is a `makedirs_safe` function in ansible.utils.path that mitigates race conditions this can run into